File: C#/euler125.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectEuler
{
    class euler125
    {
        private static bool check_pal(int num)
        {
            int temp=num;
            int rev = 0;
            if (num > 100000000)
            {
                return false;
            }
            while (temp > 0)
            {
                rev = rev * 10 + (temp % 10);
                temp = temp / 10;
            }
            if (rev == num)
            {
                return true;
            }
            return false;
        }

        public static void Main(string[] args)
        {
            var watch = new System.Diagnostics.Stopwatch();
            watch.Start();
            List<int> pal_list = new List<int>();
            long final_sum = 0;
            for (int outer = 1; outer <= Math.Sqrt(100000000); outer++)
            {
                int main = (outer * outer);
                for (int inner = outer + 1; inner <= Math.Sqrt(100000000); inner++)
                {
                    main += (inner * inner);
                    if ((check_pal(main)) && !pal_list.Contains(main))
                    {
                        final_sum += main;
                        pal_list.Add(main);
                    }
                }
            }
            watch.Stop();
            Console.WriteLine($"Required Result is: {final_sum}");
            Console.WriteLine($"Time Elapsed: {watch.ElapsedMilliseconds} ms");
            Console.ReadLine();
        }
        
    }
}

File: Python/euler125.py
import time
import math

def checkpal(n):
	if n>int(1E8):
		return False
	if n==int(str(n)[-1::-1]):
		return True
	else:
		return False

def main():
	pal_list=list()
	# for x in range(1,int(math.sqrt(int(1E8)))+1):
	# 	if checkpal(x):
	# 		pal_list.append(x)
	main=0
	final_sum=0
	# for outer in range(1,len(pal_list)):
	#  	main=(pal_list[outer]*pal_list[outer])
	#  	for inner in range(outer+1,len(pal_list)):
	#  		main+=(pal_list[inner]*pal_list[inner])
	#  		if main in pal_list:
	#  			final_sum+=main
	# print(final_sum)
	for outer in range(1,int(math.sqrt(int(1E8)))+1):
		main=(outer*outer)
		for inner in range(outer+1,int(math.sqrt(int(1E8)))+1):
			main+=(inner*inner)
			if checkpal(main) and main not in pal_list:
				final_sum+=main
				pal_list.append(main)
	print(final_sum)
	
	
if __name__ == '__main__':
	start=time.time()
	main()
	end=time.time()
	print(f"Time Elapsed: {end-start}")